Source/MIKMIDIObject.m

```diff
@@ -132,6 +140,12 @@ - (MIDIUniqueID)uniqueID
 	return _uniqueID;
 }
 
+- (void)setUniqueID:(MIDIUniqueID)uniqueID
+{
+	_uniqueID = uniqueID;
+	_hash = @(uniqueID).hash;
```

Comment:
I did not do any profiling, just went for the best performance approach, just adds a bunch of lines, so… as you wish feel free to change that ;)